src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -897,4 +904,27 @@ public void setSpectatorTarget(@Nullable org.spongepowered.api.entity.Entity ent
         return event;
     }
 
+    @Override
+    public Optional<WorldBorder> getWorldBorder() {
+        return Optional.ofNullable(this.worldBorder);
+    }
+
+    @Override
+    public void setWorldBorder(@Nullable WorldBorder border, Cause cause) {
+        if (this.worldBorder == border) {
+            return; //do not fire an event since nothing would have changed
+        }
+        if (!SpongeImpl.postEvent(SpongeEventFactory.createChangeWorldBorderEventTargetPlayer(cause, Optional.ofNullable(this.worldBorder), Optional.ofNullable(border), this))) {
+            this.worldBorder = border;
+            if (this.worldBorder != null) {
+                if (!((net.minecraft.world.border.WorldBorder) this.worldBorder).listeners.contains(this.borderListener)) {
+                    ((net.minecraft.world.border.WorldBorder) this.worldBorder).addListener(new PlayerOwnBorderListener((EntityPlayerMP) (Object) this));
+                }
+                this.connection.sendPacket(new SPacketWorldBorder((net.minecraft.world.border.WorldBorder) (Object) this.worldBorder, SPacketWorldBorder.Action.INITIALIZE));
+            } else { //unset the border if null
+                this.connection.sendPacket(new SPacketWorldBorder(new net.minecraft.world.border.WorldBorder(), SPacketWorldBorder.Action.INITIALIZE));
```

Comment:
In this case you want to initialize the world's world border again, not an empty one.